### PR TITLE
setEnabled returns future to be consistent with React Native

### PR DESCRIPTION
**Pull request by guperrot**

### CODE REVIEW

File: sdk/mobile-center-crashes/src/androidTest/java/com/microsoft/azure/mobile/crashes/CrashesAndroidTest.java [hunk 1]
```diff
@@ -312,7 +309,7 @@ public void run() {
         assertEquals(2, ErrorLogHelper.getErrorStorageDirectory().listFiles().length);
 
         /* Disable. */
-        Crashes.setEnabled(false);
+        Crashes.setEnabled(false).get();
```
Comment (jaelim-ms) [inline sdk/mobile-center-crashes/src/androidTest/java/com/microsoft/azure/mobile/crashes/CrashesAndroidTest.java]: Here too.

File: sdk/mobile-center-distribute/src/test/java/com/microsoft/azure/mobile/distribute/DistributeBeforeApiSuccessTest.java [hunk 2]
```diff
@@ -195,8 +198,17 @@ public Void answer(InvocationOnMock invocation) throws Throwable {
...
+        Distribute.setEnabled(false).thenAccept(new MobileCenterConsumer<Void>() {
+
+            @Override
+            public void accept(Void aVoid) {
+                latch.countDown();
+            }
+        });
+        assertFalse(latch.await(0, TimeUnit.MILLISECONDS));
         runnable.get().run();
+        assertTrue(latch.await(0, TimeUnit.MILLISECONDS));
```
Comment (jaelim-ms) [inline sdk/mobile-center-distribute/src/test/java/com/microsoft/azure/mobile/distribute/DistributeBeforeApiSuccessTest.java]: You are using only one count, this doesn't seem necessary.

File: sdk/mobile-center-push/src/test/java/com/microsoft/azure/mobile/push/PushTest.java [hunk 3]
```diff
@@ -177,13 +187,21 @@ public void setEnabled() {
         verify(channel).enqueue(any(PushInstallationLog.class), eq(push.getGroupName()));
 
         /* Disable. */
-        Push.setEnabled(false);
+        Push.setEnabled(false).get();
```
Comment (jaelim-ms) [inline sdk/mobile-center-push/src/test/java/com/microsoft/azure/mobile/push/PushTest.java]: Here too.

File: sdk/mobile-center-push/src/test/java/com/microsoft/azure/mobile/push/PushTest.java [hunk 4]
```diff
@@ -177,13 +187,21 @@ public void setEnabled() {
...
         assertFalse(Push.isEnabled().get());
         verify(channel).clear(push.getGroupName());
         verify(channel, times(2)).removeGroup(eq(push.getGroupName()));
 
-        /* Disable again. */
-        Push.setEnabled(false);
+        /* Disable again. Test async. */
+        final CountDownLatch latch = new CountDownLatch(1);
```
Comment (jaelim-ms) [inline sdk/mobile-center-push/src/test/java/com/microsoft/azure/mobile/push/PushTest.java]: Seems you need this CountDownLatch for onTokenRefresh to wait before verify.

File: sdk/mobile-center-analytics/src/test/java/com/microsoft/azure/mobile/analytics/AnalyticsTest.java [hunk 5]
```diff
@@ -380,7 +383,7 @@ public void setEnabled() {
         assertTrue(Analytics.isEnabled().get());
 
         /* Disable. */
-        Analytics.setEnabled(false);
+        Analytics.setEnabled(false).get();
```
Comment (guperrot) [inline sdk/mobile-center-analytics/src/test/java/com/microsoft/azure/mobile/analytics/AnalyticsTest.java]: Yeah it's just additional testing to make sure there is no deadlock or any issue by doing that as well.

File: sdk/mobile-center-analytics/src/test/java/com/microsoft/azure/mobile/analytics/AnalyticsTest.java [hunk 6]
```diff
@@ -395,7 +398,15 @@ public void setEnabled() {
...
-        Analytics.setEnabled(true);
+        final CountDownLatch latch = new CountDownLatch(1);
+        Analytics.setEnabled(true).thenAccept(new MobileCenterConsumer<Void>() {
+
+            @Override
+            public void accept(Void aVoid) {
+                latch.countDown();
+            }
+        });
+        assertTrue(latch.await(0, TimeUnit.MILLISECONDS));
```
Comment (guperrot) [inline sdk/mobile-center-analytics/src/test/java/com/microsoft/azure/mobile/analytics/AnalyticsTest.java]: It's just additional testing to make sure there is no deadlock or any issue by doing that as well.

File: sdk/mobile-center-crashes/src/androidTest/java/com/microsoft/azure/mobile/crashes/CrashesAndroidTest.java (hunk 1 above)
Comment (guperrot) [inline sdk/mobile-center-crashes/src/androidTest/java/com/microsoft/azure/mobile/crashes/CrashesAndroidTest.java]: it's just additional testing to make sure there is no deadlock or any issue by doing that as well.

File: sdk/mobile-center-distribute/src/test/java/com/microsoft/azure/mobile/distribute/DistributeBeforeApiSuccessTest.java (hunk 2 above)
Comment (guperrot) [inline sdk/mobile-center-distribute/src/test/java/com/microsoft/azure/mobile/distribute/DistributeBeforeApiSuccessTest.java]: This is to test the async behavior.

File: sdk/mobile-center-push/src/test/java/com/microsoft/azure/mobile/push/PushTest.java (hunk 3 above)
Comment (guperrot) [inline sdk/mobile-center-push/src/test/java/com/microsoft/azure/mobile/push/PushTest.java]: it's just additional testing to make sure there is no deadlock or any issue by doing that as well.

File: sdk/mobile-center-push/src/test/java/com/microsoft/azure/mobile/push/PushTest.java (hunk 4 above)
Comment (guperrot) [inline sdk/mobile-center-push/src/test/java/com/microsoft/azure/mobile/push/PushTest.java]: No, again this is a test to verify the async behavior of setEnabled

**Pull request merged**